r_thres : 100
g_thres : 100
b_thres : 100
